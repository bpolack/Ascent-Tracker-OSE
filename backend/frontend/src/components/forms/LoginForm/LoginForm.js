import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { login } from '../../../actions/authentication';
import PropTypes from 'prop-types';


export class LoginForm extends Component {
	constructor(props) {
		super(props);

		// Initial state
		this.state = {
			email: '',
			password: ''
		}

		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleChange(event) {
		event.preventDefault();
		this.setState({
			[event.target.name]: event.target.value,
		});
	}

	handleSubmit(event) {
		const { email, password } = this.state;

		event.preventDefault();
		this.props.login(email, password);
	}

	render() {

		const { isAuthenticated } = this.props;

		if (isAuthenticated) {
			return <Redirect to="/track" />;
		}

		return (
			<form onSubmit={this.handleSubmit}>
				<label className="block mb-3">
					<span className="text-ascent-std">Email Address</span>
					<input type="email" name="email" onChange={this.handleChange} placeholder="hello@world.ca" className="form-ascent" required />
				</label>

				<label className="block mb-3">
					<span className="text-ascent-std">Password</span>
					<input type="password" name="password" onChange={this.handleChange} className="form-ascent" required />
				</label>
				<input className="cursor-pointer leading-none mx-2 px-5 py-3 font-semibold transition-colors rounded-full text-white bg-indigo-500 hover:bg-pink-500" type="submit" value="Login" />
			</form>
		)
	}
}

LoginForm.propTypes = {
	login: PropTypes.func.isRequired,
	isAuthenticated: PropTypes.bool
}

const mapStateToProps = state => ({
	isAuthenticated: state.authentication.isAuthenticated
});

export default connect(mapStateToProps, { login })(LoginForm)

