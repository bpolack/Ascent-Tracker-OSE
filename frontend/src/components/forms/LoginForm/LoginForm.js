import React, { Component } from 'react'

// Bootstrap Components
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';

export class LoginForm extends Component {
	constructor(props) {
		super(props);
		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleChange(event) {
		event.preventDefault();
		this.setState({
			[event.target.name]: event.target.value,
		});
	}

	handleSubmit(event) {
		console.log('Email was submitted: ' + this.state.email);
		console.log('Password was submitted: ' + this.state.password);
		event.preventDefault();
	}

	render() {
		return (
			<Form onSubmit={this.handleSubmit}>
				<Form.Group controlId="loginEmail">
					<Form.Label>Email Address</Form.Label>
					<Form.Control type="email" name="email" onChange={this.handleChange} placeholder="hello@world.ca" required />
				</Form.Group>

				<Form.Group controlId="loginPassword">
					<Form.Label>Password</Form.Label>
					<Form.Control type="password" name="password" onChange={this.handleChange} required />
				</Form.Group>
				<Button className="px-4" variant="primary" type="submit">Login</Button>
			</Form>
		)
	}
}

export default LoginForm
