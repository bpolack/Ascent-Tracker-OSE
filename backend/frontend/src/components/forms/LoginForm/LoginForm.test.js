import React from 'react';
import { shallow } from 'enzyme';
import { LoginForm } from './LoginForm';

describe('LoginForm', () => {
	it("Renders correctly without crashing", () => {
		const loginMock = jest.fn();

		const wrapper = shallow(<LoginForm isAuthenticated={false} login={loginMock} />);
		expect(wrapper).toMatchSnapshot();
	});

	describe('When user changes input', () => {
		it('form state is updated', () => {
			const loginMock = jest.fn();
			const preventDefault = jest.fn();

			const wrapper = shallow(<LoginForm isAuthenticated={false} login={loginMock} />);

			// Change value of input field
			const inputElement = wrapper.find('input').first();
			inputElement.simulate('change', { preventDefault, target: { name: 'email', value: 'testing@testing.com' } });

			// Check that state has updated to match
			expect(wrapper.state('email')).toEqual("testing@testing.com");
		});
	});

	describe('When user submits the form', () => {
		it('details are passed to login function', () => {
			const loginMock = jest.fn();
			const preventDefault = jest.fn();

			const wrapper = shallow(<LoginForm isAuthenticated={false} login={loginMock} />);

			// Set values and simulate submit
			const formElement = wrapper.find('form');
			const nameElement = wrapper.find('input[name="email"]');
			const passElement = wrapper.find('input[name="password"]');
			nameElement.simulate('change', { preventDefault, target: { name: 'email', value: 'testing@testing.com' } });
			passElement.simulate('change', { preventDefault, target: { name: 'password', value: 'Access1234!!!' } });
			formElement.simulate('submit', { preventDefault });

			// Check that mock function called & with correct params
			expect(loginMock).toHaveBeenCalledTimes(1);
			expect(loginMock).toHaveBeenCalledWith('testing@testing.com', 'Access1234!!!');
		});
	});
});