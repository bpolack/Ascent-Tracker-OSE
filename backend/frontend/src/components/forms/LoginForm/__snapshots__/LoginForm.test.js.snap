// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Renders correctly without crashing 1`] = `ShallowWrapper {}`;
